### PR TITLE
docs(api): add instructions for how to upload files locally

### PR DESCRIPTION
**Pull request by chrisdalykinandcarta**
## Describe your changes

This is the current process as-is, it's messy and hopefully we can make lots of it obsolete.

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

**Review by lutamKC — DISMISSED** [review]

File: docs/uploading-files.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+```
+APPEALS_CASE_OFFICER_GROUP_ID
+APPEALS_INSPECTOR_GROUP_ID
+APPEALS_VALIDATION_OFFICER_GROUP_ID
+APPLICATIONS_CASE_ADMIN_OFFICER_GROUP_ID
+APPLICATIONS_CASEOFFICER_GROUP_ID
+APPLICATIONS_CASETEAM_GROUP_ID
+APPLICATIONS_INSPECTOR_GROUP_ID
+AZURE_BLOB_STORE_HOST
+AZURE_BLOB_STORE_ACCOUNT_NAME
```
Comment (beejjacobs) [inline docs/uploading-files.md:44]: This and `AZURE_BLOB_STORE_ACCOUNT_KEY` aren't required anymore

File: docs/uploading-files.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+## 3. Enable authentication and HTTPS on Web
+You'll need to enable authentication and HTTPS on the Web app. First, generate a key pair
+
+```
+openssl genrsa -out client-key.pem 2048
+openssl req -new -key client-key.pem -out client.csr
+openssl x509 -req -in client.csr -signkey client-key.pem -out client-cert.pem
+```
+
+Then place them in the root of the back-office project folder (they'll be ignored by gitignore). Then, configure your .env.local file for the web app with the following settings:
```
Comment (beejjacobs) [inline docs/uploading-files.md:57]: meant to be the web folder (or change the file paths below).

File: docs/uploading-files.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+```
+SSL_CERT_FILE=client-cert.pem
```
Comment (beejjacobs) [inline docs/uploading-files.md:60]: mention installing the cert locally to trust it, or ignoring cert errors on localhost (e.g. Edge flag)

File: docs/uploading-files.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+HTTPS_PORT=8080
+AUTH_DISABLED=false
+AUTH_CLIENT_ID={copy from portal}
+AUTH_TENANT_ID={copy from portal}
+AUTH_CLIENT_SECRET={copy from portal}
+AUTH_REDIRECT_URI=/auth/redirect
+AUTH_CLOUD_INSTANCE_ID=https://login.microsoftonline.com
+```
+
+You'll have to log into the Azure Portal and copy the values for the keys with {copy from portal} values.
```
Comment (beejjacobs) [inline docs/uploading-files.md:72]: I think AUTH_CLIENT_ID and AUTH_CLIENT_SECRET should use a different Azure App to dev (one with the localhost redirect URI).

File: docs/uploading-files.md (hunk 3 above)
Comment (beejjacobs) [inline docs/uploading-files.md:60]: mention installing the cert, or ignoring localhost cert errors (e.g. Edge flag)

File: docs/uploading-files.md (hunk 2 above)
Comment (beejjacobs) [inline docs/uploading-files.md:57]: meant to be the web folder (or change the file paths below).

File: docs/uploading-files.md (hunk 1 above)
Comment (beejjacobs) [inline docs/uploading-files.md:44]: This and AZURE_BLOB_STORE_ACCOUNT_KEY aren't required anymore

**Review by beejjacobs — CHANGES_REQUESTED** [review]
Some comments after trying this.

File: docs/uploading-files.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+If you are pointing at the dev database, you'll need to temporarily whitelist your IP on the database firewall. You can do this in the Azure Portal in the Networking tab
+
+(TODO: We're in the process of making this step redundant using VPN)
+
+## 5. Enable CORS for https://localost:8080 on the dev blob instance
+
+Navigate to 'Resource sharing (CORS)' section of the azure blob storage account you're pointing at and add https://localhost:8080 as an allowed origin for GET, OPTIONS and PUT.
+
+(TODO: This probably needs to live in Terraform for dev if we're going to make this a permanent thing)
```
[the remark this answers is not in the document]
Comment (beejjacobs) [reply docs/uploading-files.md:84]: I think this line is in relation to that: https://github.com/Planning-Inspectorate/back-office/pull/886/files#diff-055885ee41dbe4c9256d6c32947554ac3fc24ba7fdc109a72e2b523944df4665R27

Creating an `.env.local` avoids the need to make changes to commit files.

File: docs/uploading-files.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+You'll need to point your local environment at the dev environment database and blob storage. You can limit this to just the blob storage environment, but then you'll pollute the dev environment with orphaned file uploads, so it's best to point at the database too if possible.
+
+Log into the azure portal and take a look at the App Configuration settings for web, api and document api. Create a .env file in 'api' that mirrors the dev settings for:
+
+```
+DATABASE_URL
+```
+
+Then for the document API, create a .env.local file that mirrors the dev settings for:
```
Comment (DavidJWhiteDev) [inline docs/uploading-files.md:27]: There are 3 document APIs - is this the document-storage one?

**Review by chrisdalykinandcarta — COMMENTED** [review]
Resolved

**Review by beejjacobs — APPROVED** [review]
Looks good.

**Pull request merged**